### PR TITLE
Move spell_data.lua to enum/magic.lua and remove requires

### PR DESCRIPTION
**Pull request by claywar**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?
* Moves scripts/globals/spell_data.lua to scripts/enum/magic.lua to conform with naming
* Removes all requires for scripts/globals/spell_data in lua scripts
* Adds CI rule for deprecated require

## Steps to test these changes
No impact should be observed

### CODE REVIEW

File: scripts/enum/magic.lua [hunk 1]
```diff
@@ -64,22 +64,6 @@ xi.magic.element =
...
-    [xi.day.FIRESDAY]     = xi.magic.element.FIRE,
-    [xi.day.ICEDAY]       = xi.magic.element.ICE,
-    [xi.day.WINDSDAY]     = xi.magic.element.WIND,
-    [xi.day.EARTHSDAY]    = xi.magic.element.EARTH,
-    [xi.day.LIGHTNINGDAY] = xi.magic.element.THUNDER,
-    [xi.day.WATERSDAY]    = xi.magic.element.WATER,
-    [xi.day.LIGHTSDAY]    = xi.magic.element.LIGHT,
-    [xi.day.DARKSDAY]     = xi.magic.element.DARK,
-}
-
```
Comment (Xaver-DaRed) [inline scripts/enum/magic.lua:82]: Why was this moved to magic.lua? I'm guessing it's only used there

**Review by Xaver-DaRed — APPROVED** [review]

File: scripts/enum/magic.lua (hunk 1 above)
Comment (claywar) [reply scripts/enum/magic.lua:82]: Because every table in it is attached to xi.magic

Comment (claywar) [reply scripts/enum/magic.lua:82]: Oh, and that one depends on other enums
